provider-assignments/src/java/org/sakaiproject/archiver/provider/AssignmentsArchiver.java

```diff
@@ -69,22 +65,29 @@ public void destroy() {
 	@Setter
 	private AssignmentSupplementItemService assignmentSupplementItemService;
 
+	private String attachmentsHtml;
+
 	@Override
 	public void archive(final String archiveId, final String siteId, final boolean includeStudentContent) {
 
 		final List<Assignment> assignments = this.assignmentService.getListAssignmentsForContext(siteId);
 
 		for (final Assignment assignment : assignments) {
 
-			// archive the assignment data
-			final SimpleAssignment simpleAssignment = new SimpleAssignment(assignment);
-			final String html = Htmlifier.addSiteHeader(Htmlifier.toHtml(simpleAssignment),
-					this.archiverService.getSiteHeader(siteId, TOOL_ID));
-			this.archiverService.archiveContent(archiveId, siteId, TOOL_NAME, html.getBytes(), "details.html", assignment.getTitle());
+			// clear the atttachmentsHtml string in case it contains attachments from a previous assignment
+			this.attachmentsHtml = "";
```

Comment:
possible to share state as per the comment so I'd recommend this is just per assignment and passed around rather than being per entire archiver. same issue in previous PR. However I will leave this refactor for later.